src/plugins/Worker.js

```diff
@@ -43,34 +44,46 @@ class Worker extends StandardPlugin {
   async startWorker() {
     let payloadResult = null;
     const self = this;
+    const eventTypeStarting = `WORKER/${this.name.toUpperCase()}/STARTING`;
+    logger.debug(JSON.stringify({ eventTypeStarting, result: payloadResult }));
+    this.parentEvents.emit(eventTypeStarting, { type: eventTypeStarting, payload: payloadResult });
     try {
-      if (this.worker) this.stopWorker();
-      // every minutes, check the pool
-      this.worker = setInterval(this.execWorker.bind(self), this.workerIntervalTime);
+      if (this.worker) await this.stopWorker();
+
+      if (this.workerIntervalTime > 0) {
+        this.worker = setInterval(this.execWorker.bind(self), this.workerIntervalTime);
+      }
 
       if (this.executeOnStart === true) {
         if (this.onStart) {
           payloadResult = await this.onStart();
         }
       }
-      const eventType = `WORKER/${this.name.toUpperCase()}/STARTED`;
-      this.parentEvents.emit(eventType, { type: eventType, payload: payloadResult });
+      const eventTypeStarted = `WORKER/${this.name.toUpperCase()}/STARTED`;
+      logger.debug(JSON.stringify({ eventTypeStarted, result: payloadResult }));
+      this.parentEvents.emit(eventTypeStarted, { type: eventTypeStarted, payload: payloadResult });
       this.state.started = true;
 
       if (this.executeOnStart) await this.execWorker();
     } catch (err) {
+      if (console.trace) console.trace(err);
```

Comment:
Maybe it's worth to catch the trace in the error thrown?

Reply:
Catched!